### PR TITLE
Add projects filter to Performance Measurement Report

### PR DESCRIPTION
**Pull request by gigxz**
* Add project selection filter to CoC Performance Measurement Dashboard
* Switch to HudFilterBase
* Limit project selection to supported Project Types (the SPM already does this when it runs, but this makes it clear to the user which projects are included)
* Limit project selection to selected CoC
* Use describe filters as html, and update that to support `coc_code`

----

Screen Shot 2022-05-09 at 11 41 30 AM
Screen Shot 2
<img width=
022-05-09 at 11 41 44 AM" src="https://user-images.githubusercontent.com/5333982/167446725-be5a03f7-4dc9-48cb-976d-18d4122416b3.png">

### CODE REVIEW

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 1]
```diff
@@ -541,9 +546,8 @@ def can_see_client_details?(user)
         'Measure 5',
         'Measure 7',
       ]
-      # For now, we're using a fixed set of project types
+
       options = filter.to_h
-      options[:project_type_codes] = [:es, :so, :sh, :th, :ph]
```
Comment (gigxz) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb:546]: Are these actually correct for the SPM @eanders? I remember you mentioned you didn't think one of them should be there.

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 2]
```diff
@@ -151,11 +153,14 @@ def report_path_array
...
+      # report uses only one coc_code, need to adjust for the HUD filter that needs coc_codes
+      processed_filter.coc_codes = [processed_filter.coc_code]
+      processed_filter.project_ids = processed_filter.effective_project_ids
+      scope = processed_filter.apply(report_scope_source)
       scope = filter_for_range(scope)
-      scope = filter_for_coc(scope)
+
+      reset_filter
       scope
     end
```
Comment (gigxz) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb:165]: There were a couple ways to handle the `coc_code`/`coc_codes` difference. I decided to stick with the existing method of shoehorning the code into `coc_codes` as a singleton array. May be easier to discuss in a huddle if you have questions/concerns about this part @eanders

File: drivers/performance_measurement/app/models/performance_measurement/report.rb (hunk 1 above)
Comment (eanders) [reply drivers/performance_measurement/app/models/performance_measurement/report.rb:546]: Yes, you were correct, SO _is_ included.

File: drivers/performance_measurement/app/models/performance_measurement/report.rb (hunk 2 above)
Comment (eanders) [reply drivers/performance_measurement/app/models/performance_measurement/report.rb:165]: That's what I would do!

**Pull request merged**